library/goptions.ml

```diff
@@ -307,6 +346,39 @@ let declare_bool_option_and_ref ~depr ~key ~(value:bool) =
     } in
   optread
 
+let declare_string_option_and_ref ~depr ~key ~(value:string) =
+  let r_opt = ref value in
+  let optwrite v = r_opt := Option.default value v in
+  let optread () = Some !r_opt in
+  let _ = declare_stringopt_option {
+      optdepr = depr;
+      optkey = key;
+      optread; optwrite
+    } in
+  fun () -> !r_opt
+
+let declare_stringopt_option_and_ref ~depr ~key =
+  let r_opt = ref None in
+  let optwrite v = r_opt := v in
+  let optread () = !r_opt in
+  let _ = declare_stringopt_option {
+      optdepr = depr;
+      optkey = key;
+      optread; optwrite
+    } in
+  optread
+
+let declare_interpreted_string_option_and_ref ~depr ~key ~(value:'a) from_string to_string =
```

Comment:
Nit: maybe you want to label them as `~parse` / `~unparse`

Reply:
I have no preference, tho I tend to prefer `parse` as it is shorter. I'm also fine with them being unlabelled. As people say, "the one that writes the code has the power"